common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManagerSimpleCacheImpl.java

```diff
@@ -0,0 +1,190 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.cache;
+
+import android.content.Context;
+import android.os.Build;
+import android.os.SystemClock;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.google.gson.Gson;
+import com.microsoft.identity.common.internal.util.StringUtil;
+import com.microsoft.identity.common.logging.Logger;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import lombok.SneakyThrows;
+
+/**
+ * A simple metadata store definition that uses SharedPreferencesFileManager to persist, read,
+ * update, and delete data. Please note that all CRUD actions return success, as the underlying
+ * store's API does not surface a success indicator. If you need stronger guarantees that an
+ * operation was successful, use {@link SharedPreferencesSimpleCacheImpl} which is less performance
+ * oriented.
+ * <p>
+ * Data serializes as JSON.
+ *
+ * @param <T> The type of metadata that will be persisted.
+ * @see SharedPreferencesSimpleCacheImpl
+ */
+public abstract class SharedPreferencesFileManagerSimpleCacheImpl<T> implements ISimpleCache<T>, IListTypeToken {
+
+    private static final String TAG = SharedPreferencesFileManagerSimpleCacheImpl.class.getSimpleName();
+    private static final String EMTPY_ARRAY = "[]";
+    private static final String TIMING_TAG = "execWithTiming";
+
+    private ISharedPreferencesFileManager mSharedPrefsFileManager;
+    private final String mKeySingleEntry;
```

Comment:
would like to have some comments about what is this singleKey thing

Reply:
cef32cd0d38672840aec406e5bb168ce04ae481c